Editor/Scripts/CodeGenerators/GenerateRGSerializationClass.cs

```diff
@@ -0,0 +1,105 @@
+            string filePath = Path.Combine(Application.dataPath, subfolderName, fileName);
+            string fileContents = headerComment + formattedCode;
+            Directory.CreateDirectory(Path.GetDirectoryName(filePath));
+            File.WriteAllText(filePath, fileContents);
+            RGDebug.Log($"Successfully Generated {filePath}");
+            AssetDatabase.Refresh();
+        }
+
+        private static ClassDeclarationSyntax GenerateClass(List<RGActionInfo> botActions)
+        {
+            // Generate methods for deserialization based on parameter types
+            List<MemberDeclarationSyntax> methodDeclarations = new List<MemberDeclarationSyntax>();
+            HashSet<string> processedTypes = new HashSet<string>();
+
+            foreach (RGActionInfo botAction in botActions)
+            {
+                foreach (RGParameterInfo parameter in botAction.Parameters)
+                {
+                    if (RGUtils.IsCSharpPrimitive(parameter.Type))
+                    {
+                        continue;
+                    }
+                    
+                    if (!processedTypes.Contains(parameter.Type))
+                    {
+                        processedTypes.Add(parameter.Type);
+
+                        // Generate method
+                        MethodDeclarationSyntax method = SyntaxFactory
+                            .MethodDeclaration(SyntaxFactory.ParseTypeName(parameter.Type),
+                                $"Deserialize_{parameter.Type.Replace(".", "_")}")
+                            .AddModifiers(SyntaxFactory.Token(SyntaxKind.PublicKeyword),
+                                SyntaxFactory.Token(SyntaxKind.StaticKeyword))
+                            .AddParameterListParameters(SyntaxFactory.Parameter(SyntaxFactory.Identifier("paramJson"))
+                                .WithType(SyntaxFactory.ParseTypeName("string")))
+                            .WithBody(SyntaxFactory.Block(SyntaxFactory.ReturnStatement(
+                                SyntaxFactory.InvocationExpression(
+                                    SyntaxFactory.MemberAccessExpression(
+                                        SyntaxKind.SimpleMemberAccessExpression,
+                                        SyntaxFactory.ParseTypeName("JsonUtility"),
+                                        SyntaxFactory.GenericName(SyntaxFactory.Identifier("FromJson"))
+                                            .WithTypeArgumentList(SyntaxFactory.TypeArgumentList(
+                                                SyntaxFactory.SingletonSeparatedList<TypeSyntax>(
+                                                    SyntaxFactory.ParseTypeName(parameter.Type))))),
+                                    SyntaxFactory.ArgumentList(SyntaxFactory.SingletonSeparatedList(
+                                        SyntaxFactory.Argument(SyntaxFactory.IdentifierName("paramJson"))))))));
+
+                        methodDeclarations.Add(method);
```

Comment:
Little hard to follow this - might be useful to add a comment that shows an example of what may be generated here

Reply:
Added 👍